test/unit-test/dawn/Optimizer/Passes/TestFieldAccessIntervals.cpp

```diff
@@ -0,0 +1,184 @@
+//===--------------------------------------------------------------------------------*- C++ -*-===//
```

Comment:
as a general note, I had to change this kind of setup slightly for boundary condition checks and am now returning a unique-ptr to the ISIR rather then the vector of stencils. Maybe we should be doing that throughout?

Reply:
done